cpp/models/ode_seir/model.h

```diff
@@ -38,35 +41,37 @@ namespace oseir
 {
 
 /********************
-    * define the model *
-    ********************/
+ * define the model *
+ ********************/
 
-class Model : public CompartmentalModel<InfectionState, Populations<InfectionState>, Parameters>
+// clang-format off
+using Flows = TypeList<Flow<InfectionState::Susceptible, InfectionState::Exposed>,
+                       Flow<InfectionState::Exposed,     InfectionState::Infected>,
+                       Flow<InfectionState::Infected,    InfectionState::Recovered>>;
+// clang-format on
+
+class Model : public FlowModel<InfectionState, Populations<InfectionState>, Parameters, Flows>
 {
-    using Base = CompartmentalModel<InfectionState, mio::Populations<InfectionState>, Parameters>;
+    using Base = FlowModel<InfectionState, mio::Populations<InfectionState>, Parameters, Flows>;
 
 public:
     Model()
         : Base(Populations({InfectionState::Count}, 0.), ParameterSet())
     {
     }
 
-    void get_derivatives(Eigen::Ref<const Eigen::VectorXd> pop, Eigen::Ref<const Eigen::VectorXd> y, double t,
-                         Eigen::Ref<Eigen::VectorXd> dydt) const override
+    void get_flows(Eigen::Ref<const Eigen::VectorXd> pop, Eigen::Ref<const Eigen::VectorXd> y, double t,
+                   Eigen::Ref<Eigen::VectorXd> flows) const override
     {
         auto& params     = this->parameters;
         double coeffStoE = params.get<ContactPatterns>().get_matrix_at(t)(0, 0) *
                            params.get<TransmissionProbabilityOnContact>() / populations.get_total();
 
-        dydt[(size_t)InfectionState::Susceptible] =
-            -coeffStoE * y[(size_t)InfectionState::Susceptible] * pop[(size_t)InfectionState::Infected];
-        dydt[(size_t)InfectionState::Exposed] =
-            coeffStoE * y[(size_t)InfectionState::Susceptible] * pop[(size_t)InfectionState::Infected] -
+        flows[get_flat_flow_index<InfectionState::Susceptible, InfectionState::Exposed>()] =
+            coeffStoE * y[(size_t)InfectionState::Susceptible] * pop[(size_t)InfectionState::Infected];
```

Comment:
If we decide to make a matrix or vector based on Eigen::Matrix<ScalarType, ...>, we could add something like an operator[](enum) or operator[](index) to make these lines shorter. 
However, I think this is out of scope for this pr. But we should revisit this idea, since making these lines shorter will also make the get_derivatives/get_flows functions easier to read/write.